tsp-output/@azure-tools/typespec-csharp/src/Generated/Models/CreateChatCompletionRequest.Serialization.cs

```diff
@@ -10,17 +10,24 @@
 
 namespace OpenAI.Models
 {
-    public partial class CreateChatCompletionRequest : IJsonModel<CreateChatCompletionRequest>
+    public partial class CreateChatCompletionRequest : JsonModel<CreateChatCompletionRequest>
     {
-        void IJsonModel<CreateChatCompletionRequest>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        protected override void WriteCore(Utf8JsonWriter writer, ModelReaderWriterOptions options)
         {
-            var format = options.Format == "W" ? ((IPersistableModel<CreateChatCompletionRequest>)this).GetFormatFromOptions(options) : options.Format;
-            if (format != "J")
+            // Note: support format "W*" that doesn't write start/end object 
+            // to enable extending write routine.
+            var format = (options.Format == "W" || options.Format == "W*") ? ((IPersistableModel<CreateChatCompletionRequest>)this).GetFormatFromOptions(options) : options.Format;
+            if (format != "J" && format != "J*")
             {
                 throw new FormatException($"The model {nameof(CreateChatCompletionRequest)} does not support writing '{format}' format.");
             }
 
-            writer.WriteStartObject();
+            if (format == "J")
```

Comment:
Allow writing partial content instead of full object to enable writing content with additions in the extended type.